File: 실습/6장/08_associate/Main.java
import java.util.Arrays;
import java.util.List;
import java.util.Map;
import java.util.function.Function;
import java.util.stream.Collectors;

public class Main {
    static class Person {
        private final String name;
        private final int age;

        Person(String name, int age) {
            this.name = name;
            this.age = age;
        }

        public String getName() {
            return name;
        }

        public int getAge() {
            return age;
        }

        @Override
        public String toString() {
            return name + "(" + age + ")";
        }
    }

    public static void main(String[] args) {
        List<Person> people = Arrays.asList(
                new Person("철수", 20),
                new Person("영희", 21)
        );

        Map<String, Person> peopleByName =
                people.stream().collect(
                        Collectors.toMap(
                                Person::getName,
                                Function.identity()
                        )
                );

        Map<String, Integer> nameToAge =
                people.stream().collect(
                        Collectors.toMap(
                                Person::getName,
                                Person::getAge
                        )
                );

        System.out.println(peopleByName);
        System.out.println(nameToAge);
    }
}

File: 실습/6장/08_associate/Main.kt
data class Person(
    val name: String,
    val age: Int
)

fun main() {
    val people = listOf(
        Person("철수", 20),
        Person("영희", 21)
    )

    val peopleByName = people.associateBy {
        it.name
    }

    val nameToAge = people.associate {
        it.name to it.age
    }

    println(peopleByName)
    println(nameToAge)
}